### PR TITLE
Milestone 3: set up associations

### PR DESCRIPTION
**Pull request by jaflih**
### Milestone 3: set up associations

- Created Classroom class
- Created the has-many/belongs-to relationship between Classroom and Student
- Created Rental and Book class
- Create the many-to-many relationship between Person and Book using the intermediate class Rental

### CODE REVIEW

File: lib/book.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rentals(rental)
+    @rentals.push(rental)
+    rental.book = self
+  end
```
Comment (FrederickMih) [inline lib/book.rb:15]: - [ ] Great job for implementing this method of adding rentals :+1: . Please let's also make sure that a book should be rented out to a person on a particular date. Hence you will be required to pass in `person` and `date` as a parameter in the `add_rentals` method and to achieve the has_many association, the book itself should be sent as a parameter to create a new rental ie `Rental.new(date, self, person)`. Please let's fix this issue and be sure to receive the comment `Status: Approved` from the next reviewer.

File: lib/rental.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
+require_relative 'book'
+require_relative 'person'
+
+class Rental
+  attr_accessor :date
+  attr_reader :book, :person
```
Comment (FrederickMih) [inline lib/rental.rb:6]: - [ ] In the Rental class, a book and a person can also be written. So you will need Getters and Setters methods for the @book and @person instance variables ie `attr_accessor :date, :book, :person`

File: lib/person.rb [hunk 3]
```diff
@@ -20,6 +22,11 @@ def correct_name
     @name
   end
 
+  def add_rentals(rental)
+    @rentals.push(rental)
+    rental.person = self
+  end
```
Comment (FrederickMih) [inline lib/person.rb:28]: - [ ] A book at a particular date is rented out to a person hence you will be required to pass in the `book` and `date` as a parameter to this method and continued as explained above.


  Thanks andnI wish you the best :pray:

**Review by FrederickMih — CHANGES_REQUESTED** [review]
Hi @jaflih , 

You have done a fantastic job so far in this project. Keep up! :rocket: 
There are some issues that you still need to work on to go to the next project but you are almost there!


# To Highlights

-  Git/GitHub workflow has been followed :heavy_check_mark: 
-  Linters are correctly set up :heavy_check_mark: 
-  Pull Request has a title and summary :heavy_check_mark: 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions


_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: lib/classroom.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
+require_relative 'student'
+
+class Classroom
+  attr_accessor :label, :students
```
[the remark this answers is not in the document]
Comment (jaflih) [reply lib/classroom.rb:4]: Hi @FrederickMih, thank you for your feedback.
It's done.

File: lib/rental.rb (hunk 2 above)
Comment (jaflih) [reply lib/rental.rb:6]: It's done.

File: lib/person.rb (hunk 3 above)
Comment (jaflih) [reply lib/person.rb:28]: Ok done

File: lib/book.rb (hunk 1 above)
Comment (jaflih) [reply lib/book.rb]: ok thank you

**Review by t-yanick — APPROVED** [review]
Hi @jaflih, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by jaflih** [comment]
@t-yanick 
thanks

**Pull request merged**